File: app/src/main/java/com/example/myapplication/ui/recyclerView/StaggeredGridRecyclerViewActivity.java
package com.example.myapplication.ui.recyclerView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import android.os.Bundle;

import com.example.myapplication.R;

public class StaggeredGridRecyclerViewActivity extends AppCompatActivity {
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_staggered_grid);
        recyclerView = findViewById(R.id.staggered_recyclerview);
        recyclerView.setLayoutManager(new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL));
        recyclerView.setAdapter(new StaggeredGridRecyclerViewAdapter(this));
    }
}

File: app/src/main/java/com/example/myapplication/ui/recyclerView/StaggeredGridRecyclerViewActivity.kt
package com.example.myapplication.ui.recyclerView

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import com.example.myapplication.R

class StaggeredGridRecyclerViewActivity : AppCompatActivity() {
    private var recyclerView: RecyclerView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_staggered_grid)
        recyclerView = findViewById(R.id.staggered_recyclerview)
        recyclerView?.layoutManager = StaggeredGridLayoutManager(
            2,
            StaggeredGridLayoutManager.VERTICAL
        )
        recyclerView?.adapter = StaggeredGridRecyclerViewAdapter(this)
    }
}